Added www.newpct.com provider

CristianBB opened the pull request:
Added www.newpct.com provider, prepared to seek series with Spanish
dubbing.
I've added two new options on provider's config:

- For spanish torrents: This option allows to make show searchs from
this provider ONLY if "show's config => Info Language" is marked as
Spanish. In this way we avoid search from this supplier for shows that
we have configured in VOs or other languages.

- Append identifier: Append an identifier to every episode snatched by
this provider. Usefull in combination with "Required Words" on show's
configuration if you want to download certain shows only from this
provider. In this way, if we have other providers, we avoid to snatch
torrents from other providers to prevent that episodes are downloaded in
other languages, so it is necessary that "Required Words" on show's
configuration are the same that we define here.

I hope that the changes are of your pleasure :)

CristianBB

SiCKRAGEB0T commented:
Thanks for the pull request! Before a real human comes by, please make sure your report has all the below criteria checked

- [x] Give a description on what the PR is for.
- [x] Make sure you send a PR against the DEVELOP branch
- [x] Don't send big changes in one go. Split up big PRs into multiple smaller (easier manageble) PRs

Please make sure you also read [contribution guide](https://github.com/SiCKRAGETV/SickRage/blob/master/contributing.md) and followed all the steps.

Thanks!

miigotu commented:
Few issues I have with this one.

First, why override _get_episode search strings? This can surely be avoided somehow, and it is highly discouraged to override the search strings. You will have to to a replace on the search string in _doSearch for both searching and the results. temporada and capitulo just wont do to match in post processing.

Second, not only does this provider provide size which is not parsed into the item result, the provider does not provide seeds, leechers, hash, which makes itm highly undesirable as a provider in sickrage

CristianBB commented:
Hi miigotu,
In response to your questions:

- After many test with www.newpct.com search, it does not correctly process the search by episode number (always returns all results to be reported regardless of the number of episode). you can do the test by entering "The strain" and "the strain 1x01", which returns results have nothing to do with the informed chapter ... In fact I had to implement an algorithm to form a chain type: show name + season + episode so SickRage can recognize results returned (newpct shows are typed as: The Strain - Season 1 [HDTV] [Cap.109] [Spanish Castilian] and none of the implemented algorithms can recognize it).

- I have checked and I can include the process of file size as the page returned as results, but it does not report the number of seeds and leechers so that function could not be implemented ...

It really is a shame, but the Spanish trackers have a very particular way of naming the series, so I added the option to only use this provider in the series that you want to download in Spanish to avoid an extra proccess.

miigotu commented:
If it doesnt make a different whether 1x1 or w/e is there then there is no reason to override _get_episode_search strings right?

CristianBB commented:
Ok course, 
I just thought there would be less chances of getting invalid results if removed the rest of the search string

Allow me a few minutes and will remove the code _get_episode_search

CristianBB commented:
Deleted,
I'll try to parse the torrent size this weekend, please tell me anything.

I have pending a change to add a new regular expression on file "regexes" that I'd like to discuss with you, to parse correctly filenames with format: "Show Name - Random text. [Quality]. [Cap.102].Random text"

miigotu commented:
Adding a new regex would be the best solution I think, even though I hate touching the regex that usually work ok as is =P

CristianBB commented:
Hi miigotu,
I've been working hard on that, let me tell you:

- Finally I've implemented the new regexe to process correctly show names with format: "Show Name - Random text. [Quality]. [Cap.102].Random text"
It's working OK on position that I've used (cause of scoring system), even parsing files with autoprocess and manual proccess.

- I've added torrent's size.

- There is a change that you should review, on file  "sickbeard/providers/generic.py", in "findSearchResults" code, when it's parsing the showname I've changed:
myParser = NameParser(False)
To 
myParser = NameParser(showObj=show)

Original code causes that NamePaser compile all regexes available including anime  (cause show object was not being informed). Is there any reason to do this? That algorithm sometimes returns the result of an anime regexe even if show analyzed is not anime. Please check if my change is ok, otherwise I should have to overwrite "findSearchResults" on newpct provider.

Anything please let me know :)

Regards,
Cristian.

miigotu commented on sickbeard/providers/newpct.py:
```diff
@@ -0,0 +1,201 @@
+            size = size * 1024**4
+        return size
+
+    def _processTittle(self, title):
```
mispelled =P

Is this quality borking absolutely necessary?

CristianBB replied:
Ups O_O, fixed

miigotu commented on sickbeard/providers/newpct.py:
```diff
@@ -0,0 +1,201 @@
+from sickbeard.common import USER_AGENT
+from sickbeard.bs4_parser import BS4Parser
+from sickbeard import scene_exceptions
+from ctypes.test.test_sizes import SizesTestCase
```
unused import

miigotu commented on sickbeard/providers/newpct.py:
```diff
@@ -0,0 +1,201 @@
+from sickbeard.providers import generic
+from sickbeard.common import USER_AGENT
+from sickbeard.bs4_parser import BS4Parser
+from sickbeard import scene_exceptions
```
unused import

miigotu commented on sickbeard/providers/newpct.py:
```diff
@@ -0,0 +1,201 @@
+# You should have received a copy of the GNU General Public License
+# along with SickRage.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import with_statement
```
not supporting python 2.6, so no need to import with from future

miigotu commented on sickbeard/providers/generic.py:
```diff
@@ -384,7 +384,7 @@ def findSearchResults(self, show, episodes, search_mode, manualSearch=False, dow
             # parse the file name
             try:
-                myParser = NameParser(False)
+                myParser = NameParser(showObj=show)
```
This is good, we should always pass the show object to NameParser if we have it.

CristianBB commented:
Sorry for that, fixed =)

miigotu commented:
Ok, everything looks pretty ok now, just conflicts and rebase. I can do that manually later.

CristianBB commented:
Ok miigotu, if you need me to do something let me know.
